### PR TITLE
RNASeQC with plots, fastq combining and additional bugfixes

### PR DESCRIPTION
**Pull request by Gerbenvandervries**

### CODE REVIEW

File: protocols/CopyToResultsDir.sh [hunk 1]
```diff
@@ -53,68 +53,68 @@ mkdir -p "${projectResultsDir}/qcmetrics"
...
 # Copy variant vcfs.
 
-        cp "${projectBatchGenotypedVIPPrefix}"* "${projectResultsDir}/variants/vip/"
-	cp "${projectBatchGenotypedVariantCalls}"* "${projectResultsDir}/variants/"
+        rsync -av "${projectBatchGenotypedVIPPrefix}"* "${projectResultsDir}/variants/vip/"
+	rsync -av "${projectBatchGenotypedVariantCalls}"* "${projectResultsDir}/variants/"
 
 # Copy leafcutter
-	cp "${intermediateDir}"*.leafcutter.outlier* "${projectResultsDir}/leafcutter/"
+	rsync -av "${intermediateDir}"*.leafcutter.outlier* "${projectResultsDir}/leafcutter/"
```
Comment (RoanKanninga) [inline protocols/CopyToResultsDir.sh]: I would put a slash behind the ${intermediateDir}, just to be sure

File: protocols/TrimReads.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+#Echo parameter values
+echo "seqType: ${seqType}"
+echo "peEnd1BarcodeFqGz: ${peEnd1BarcodeFqGz}"
+echo "peEnd2BarcodeFqGz: ${peEnd2BarcodeFqGz}"
+echo "srBarcodeFqGz: ${srBarcodeFqGz}"
+echo "intermediateDir: ${intermediateDir}"
+
+#Load module
+module load ${BBMapVersion}
```
Comment (RoanKanninga) [inline protocols/TrimReads.sh]: quotes

File: protocols/TrimReads.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	ref=${EBROOTBBMAP}/resources/polyA.fa.gz,${EBROOTBBMAP}/resources/truseq.fa.gz \
+	overwrite=true \
+        k=13 \
+	ktrim=l \
+        qtrim=rl \
+        trimq=14 \
+        minlength=20 > ${intermediateDir}/${externalSampleID}_L${lane}.trimming.log 2>&1
+	#\forcetrimleft=11
+
+	gzip ${peEnd1BarcodeFqGz}.tmp
```
Comment (RoanKanninga) [inline protocols/TrimReads.sh]: quotes

File: protocols/TrimReads.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	ktrim=r \
+	useshortkmers=t \
+	mink=5 \
+	qtrim=r \
+	trimq=10 \
+        minlength=20 > ${intermediateDir}/${externalSampleID}_L${lane}.trimming.log 2>&1
+
+k=13 ktrim=r useshortkmers=t mink=5 qtrim=r trimq=10 minlength=20
+	gzip ${srBarcodeFqGz}.tmp
+	mv ${srBarcodeFqGz}.tmp.gz ${srBarcodeFqGz}
```
Comment (RoanKanninga) [inline protocols/TrimReads.sh]: quotes

File: test/test_pipeline.sh [hunk 5]
```diff
@@ -1,209 +1,70 @@
...
-	echo "copy generate template"
-	cp ${workfolder}/tmp/NGS_RNA/templates/generate_template.sh ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
-
-	#fgrep "computeVersion," ${workfolder}/tmp/NGS_RNA/parameters.csv > ${workfolder}/generatedscripts/${_projectName}/mcVersion.txt
-
-##############
-perl -pi -e "s|module load NGS_DNA|module load NGS_DNA/betaAutotest|" ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
-###### Load a version of molgenis compute
-sudo -u umcg-envsync bash -l << EOF
+	#rsync -r --verbose --recursive --links --no-perms --times --group --no-owner --devices --specials ${TMPHOME}/test/rawdata/MY_TEST_BAM_PROJECT/SRR1552906[249]_[12].fq.gz ${workfolder}/rawdata/ngs/MY_TEST_BAM_PROJECT/
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: kan weg

File: test/test_pipeline.sh [hunk 6]
```diff
@@ -1,209 +1,70 @@
...
+	rsync -r --verbose --recursive --links --no-perms --times --group --no-owner --devices --specials ${pipelinefolder}/test/rawdata/MY_TEST_BAM_PROJECT/SRR1552906[249]_[12].fq.gz ${workfolder}/rawdata/ngs/MY_TEST_BAM_PROJECT/
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: quotes

File: test/test_pipeline.sh [hunk 7]
```diff
@@ -1,209 +1,70 @@
...
 
-id
-export SOURCE_HPC_ENV="True"
-. ~/.bashrc
-module load depad-utils
-#module list 
-hpc-environment-sync.bash -m NGS_RNA/betaAutotest
-exit
+	rm -rf ${workfolder}/{tmp,generatedscripts,projects}/NGS_RNA/${_projectName}/
+	mkdir -p ${workfolder}/generatedscripts/${_projectName}/
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: quotes

File: test/test_pipeline.sh [hunk 8]
```diff
@@ -1,209 +1,70 @@
...
 
-EOF
+	echo "copy generate template"
+	cp ${pipelinefolder}/templates/generate_template.sh ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: quotes

File: test/test_pipeline.sh [hunk 9]
```diff
@@ -1,209 +1,70 @@
...
-	perl -pi -e "s|module load Molgenis-Compute/dummy|module load Molgenis-Compute/\$mcVersion|" ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
-
+	#perl -pi -e "s|module load ${NGS_RNA_VERSION}|EBROOTNGS_RNA=${workfolder}/tmp/NGS_RNA/|" ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
+	#perl -pi -e 's|create_in-house_ngs_projects_workflow.csv|create_external_samples_ngs_projects_workflow.csv|' ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
+	#perl -pi -e 's|ngsversion=.*|ngsversion="test";\\|' ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
+	#perl -pi -e 's|sh \$EBROOTMOLGENISMINCOMPUTE/molgenis_compute.sh|module load Molgenis-Compute/dummy\nsh \$EBROOTMOLGENISMINCOMPUTE/molgenis_compute.sh|' ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
+	#perl -pi -e "s|module load Molgenis-Compute/dummy|module load Molgenis-Compute/\$mcVersion|" ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: kan weg?

File: test/test_pipeline.sh [hunk 10]
```diff
@@ -1,209 +1,70 @@
...
 	perl -pi -e 's|WORKFLOW=\${EBROOTNGS_RNA}/workflow_\${PIPELINE}.csv|WORKFLOW=\${EBROOTNGS_RNA}/test_workflow_\${PIPELINE}.csv|' ${workfolder}/generatedscripts/${_projectName}/generate_template.sh
-	cp ${workfolder}/tmp/NGS_RNA/test/${_projectName}.csv ${workfolder}/generatedscripts/${_projectName}/
-	perl -pi -e "s|/groups/umcg-atd/tmp03/|${workfolder}/|g" ${workfolder}/generatedscripts/${_projectName}/${_projectName}.csv
+
+	cp "${pipelinefolder}/test/${_projectName}.csv" "${_generatedScriptsFolder}"
+	perl -p -e "s|/groups/umcg-atd/tmp01/|${workfolder}/|g" "${_generatedScriptsFolder}/${_projectName}.csv" > "${_generatedScriptsFolder}/${_projectName}.csv.tmp"
+	mv -v "${_generatedScriptsFolder}/${_projectName}.csv"{.tmp,}
+
+	#perl -pi -e "s|/groups/umcg-atd/tmp03/|${workfolder}/|g" ${workfolder}/generatedscripts/${_projectName}/${_projectName}.csv
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: kan weg?

File: test/test_pipeline.sh [hunk 11]
```diff
@@ -1,209 +1,70 @@
...
+
 	cd ${workfolder}/generatedscripts/${_projectName}/
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: quotes

File: test/test_pipeline.sh [hunk 12]
```diff
@@ -1,209 +1,70 @@
...
-	perl -pi -e "s|/apps/software/${NGS_RNA_VERSION}/|${workfolder}/tmp/NGS_RNA/|g" *.sh
 
-	perl -pi -e 's|slurm/header_tnt.ftl|slurm/header.ftl|' *.sh
-	perl -pi -e 's|slurm/footer_tnt.ftl|slurm/footer.ftl|' *.sh
+	###### Load a version of molgenis compute
+	perl -pi -e "s|workflow_STAR.csv|test_workflow_STAR.csv|" *.sh
+#	perl -pi -e "s|/apps/software/${NGS_RNA_VERSION}/|${workfolder}/tmp/NGS_RNA/|g" *.sh
 
+#	perl -pi -e 's|slurm/header_tnt.ftl|slurm/header.ftl|' *.sh
+#	perl -pi -e 's|slurm/footer_tnt.ftl|slurm/footer.ftl|' *.sh
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: kan weg?

File: test/test_pipeline.sh [hunk 13]
```diff
@@ -1,209 +1,70 @@
...
 
 	sh submit.sh
 
-	echo "cd ${workfolder}/projects/${_projectName}/run01/jobs/"
 	cd ${workfolder}/projects/${_projectName}/run01/jobs/
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: quotes

File: test/test_pipeline.sh [hunk 14]
```diff
@@ -234,65 +95,55 @@ function checkIfFinished(){
...
+mkdir -p "${pipelinefolder}"
 cd "${pipelinefolder}"
 
-echo "pr number: $1"
-
-PULLREQUEST=$1
-NGS_RNA_VERSION=NGS_RNA/3.3.0
+#echo "pr number: ${1}"
 
+PULLREQUEST=35
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: moet straks $1 worden

File: test/test_pipeline.sh [hunk 15]
```diff
@@ -234,65 +95,55 @@ function checkIfFinished(){
...
-=======
-### create testworkflow
-cd ${workfolder}/tmp/NGS_RNA/
-cp ${workfolder}/tmp/NGS_RNA/workflow_hisat.csv ${workfolder}/tmp/NGS_RNA/test_workflow_hisat.csv 
-tail -1 ${workfolder}/tmp/NGS_RNA/workflow_hisat.csv | perl -p -e 's|,|\t|g' | awk '{print "Autotest,test/protocols/Autotest.sh,"$1}' >> ${workfolder}/tmp/NGS_RNA/test_workflow_hisat.csv
+#exclude steps.
+perl -pi -e 's|s09_OUTRIDER|#s09_OUTRIDER|g' test_workflow_STAR.csv
+perl -pi -e 's|s12_VIP|#s12_VIP|g' test_workflow_STAR.csv
+#cp ${workfolder}/tmp/NGS_RNA/test/results/PlatinumSubset_NGS_RNA.variant.calls.genotyped.chr1.true.vcf /home/umcg-molgenis/NGS_RNA/
+#cp ${workfolder}/tmp/NGS_RNA/test/results/PlatinumSubset_NGS_RNA.expression.true.table /home/umcg-molgenis/NGS_RNA/
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: kan weg?

File: test/test_pipeline.sh [hunk 16]
```diff
@@ -234,65 +95,55 @@ function checkIfFinished(){
...
 
-cp ${workfolder}/tmp/NGS_RNA/test/results/PlatinumSubset_NGS_RNA.variant.calls.genotyped.chr1.true.vcf /home/umcg-molgenis/NGS_RNA/
-cp ${workfolder}/tmp/NGS_RNA/test/results/PlatinumSubset_NGS_RNA.expression.true.table /home/umcg-molgenis/NGS_RNA/
+#cp test/results/* /groups/umcg-atd/tmp01/tmp/NGS_RNA/testdata_true/
+cp /home/umcg-gvdvries/git/NGS_RNA/test/results/* /groups/umcg-atd/tmp01/tmp/NGS_RNA/testdata_true/
```
Comment (RoanKanninga) [inline test/test_pipeline.sh]: quotes

File: protocols/TrimReads.sh (hunk 2 above)
Comment (Gerbenvandervries) [reply protocols/TrimReads.sh]: moved to deprecated/

File: protocols/CopyToResultsDir.sh (hunk 1 above)
Comment (Gerbenvandervries) [reply protocols/CopyToResultsDir.sh]: done

File: test/test_pipeline.sh (hunk 14 above)
Comment (Gerbenvandervries) [reply test/test_pipeline.sh]: ow ja:)

**Review by RoanKanninga — APPROVED** [review]

**Pull request merged**